prometheus/promhttp/instrument_client.go

```diff
@@ -78,7 +78,11 @@ func InstrumentRoundTripperCounter(counter *prometheus.CounterVec, next http.Rou
 			for label, resolve := range rtOpts.extraLabelsFromCtx {
 				l[label] = resolve(resp.Request.Context())
 			}
-			counter.With(l).(prometheus.ExemplarAdder).AddWithExemplar(1, rtOpts.getExemplarFn(r.Context()))
+			if rtOpts.getExemplarFn == nil {
+				counter.With(l).Inc()
+			} else {
+				counter.With(l).(prometheus.ExemplarAdder).AddWithExemplar(1, rtOpts.getExemplarFn(r.Context()))
+			}
```

Comment:
nit: without modifying `getExemplarFn` from `defaultOptions`, you could check if returned labels are `nil`.

```suggestion
			exemplarLabels := rtOpts.getExemplarFn(r.Context())
			if exemplarLabels == nil {
				counter.With(l).Inc()
			} else {
				counter.With(l).(prometheus.ExemplarAdder).AddWithExemplar(1, exemplarLabels)
			}
```